app/src/main/java/ai/elimu/appstore/BaseApplication.java

```diff
@@ -47,24 +68,25 @@ protected void log(int priority, String tag, String message, Throwable throwable
         Timber.i("onCreate");
 
         // greenDAO config
-        CustomDaoMaster.DevOpenHelper helper = new CustomDaoMaster.DevOpenHelper(this, "appstore-db");
+        CustomDaoMaster.DevOpenHelper helper = new CustomDaoMaster.DevOpenHelper(this,
+                "appstore-db");
         Database db = helper.getWritableDb();
         daoSession = new CustomDaoMaster(db).newSession();
 
         // Check if the application's versionCode was upgraded
-        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(getApplicationContext());
-        int oldVersionCode = sharedPreferences.getInt(PREF_APP_VERSION_CODE, 0);
+        int oldVersionCode = AppPrefs.getAppVersionCode();
         int newVersionCode = VersionHelper.getAppVersionCode(getApplicationContext());
         if (oldVersionCode == 0) {
-            sharedPreferences.edit().putInt(PREF_APP_VERSION_CODE, newVersionCode).commit();
+            AppPrefs.saveAppVersionCode(newVersionCode);
```

Comment:
Good idea to refactor this logic into a separate file :-)
It makes the code easier to read and maintain.

Reply:
Yes done it.